db/migrate/20191023180308_subclass_ems_cluster.rb

```diff
@@ -0,0 +1,39 @@
+class SubclassEmsCluster < ActiveRecord::Migration[5.1]
+  class ExtManagementSystem < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class EmsCluster < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    [
+      'Kubevirt',
+      'Openstack',
+      'Microsoft',
+      'Redhat',
+      'Vmware',
+    ].each do |provider|
+      ems_class_name = "ManageIQ::Providers::#{provider}::InfraManager"
+      cluster_class_name = "#{ems_class_name}::Cluster"
+      EmsCluster.in_my_region.
+        joins('join ext_management_systems on ems_clusters.ems_id = ext_management_systems.id').
+        where(:ext_management_systems => {:type => ems_class_name}).
```

Comment:
Since you have both the ems_cluster stub and the ext_management_system stub, you can just use a normal has_many relationship...

In EmsCluster stub:

```ruby
belongs_to :ext_management_system, :class_name => "SubclassEmsCluster::ExtManagementSystem"
```

Then here do:

```ruby
EmsCluster.in_my_region
  .where(:ext_management_system => {:type => ems_class_name})
  .update_all(:type => cluster_class_name)
```